### ISSUE TITLE
sftp prompt broken

### ISSUE THREAD

**Issue by cw789** [git-for-windows/git#2257]
- [x] I was not able to find an [open](https://github.com/git-for-windows/git/issues?q=is%3Aopen) or [closed](https://github.com/git-for-windows/git/issues?q=is%3Aclosed) issue matching what I'm seeing

### Setup

 - Which version of Git for Windows are you using? Is it 32-bit or 64-bit?

```
$ git --version --build-options

git version 2.22.0.windows.1
cpu: x86_64
built from commit: d003d728ffa6c0006da875ec6318d3f6b28a4ddb                                                             sizeof-long: 4
sizeof-size_t: 8
```

 - Which version of Windows are you running? Vista, 7, 8, 10? Is it 32-bit or 64-bit?

```
$ cmd.exe /c ver

Microsoft Windows [Version 10.0.18362.239]
```

### Details

 - Which terminal/shell are you running Git from? e.g Bash/CMD/PowerShell/other

CMD & Pwsh

 - What commands did you run to trigger this issue? If you can provide a
   [Minimal, Complete, and Verifiable example](http://stackoverflow.com/help/mcve)
   this will help us understand the issue.

```
sftp hostname
```
 - What did you expect to occur after running these commands?

<pre>sftp>_</pre>

 - What actually happened instead?
<pre>s     _</pre>

**Comment by rimrul** [comment]
Are you sure this involves Git for Windows in any way? The `sftp.exe` shipped with Git for Windows 2.22.0 prints it's usage in Git Bash, Powershell and CMD when called without arguments. @dscho: Why does Git for Windows ship `sftp.exe` anyways? I can't find any place where we use it or support the usage of sftp. We exclude it for MinGit, but ship it for everything else. Can we exclude it? Should we keep it?

**Comment by cw789** [comment]
I'm pretty sure it's caused by the shipped `sftp.exe`.
Am I right if I assume this then involves `MSYS2-packages - openssh`?

**Comment by rimrul** [comment]
Could you please check that using `where sftp` (CMD) or `get-command sftp` (Powershell)? If it indeed is the `sftp.exe` shipped with Git for windows, then yes, it should be from the OpenSSH MSYS2 package. But OpenSSH's SFTP should expect a hostname argument and print it's usage if none is supplied. Neither your expected output nor your actual output look like that.

**Comment by cw789** [comment]
Yeah, it is `sftp.exe` shipped with Git for Windows.
Indeed I just forgot to put the `hostname` into my description.

**Comment by dscho** [comment]
> Why does Git for Windows ship `sftp.exe` anyways?

Because it is part of the `openssh` package, and we do not explicitly exclude that file (except [in MinGit's case](https://github.com/git-for-windows/build-extra/blob/9b05df7752b8b5e2bc2a769af661d1966d6a2a63/make-file-list.sh#L225)).

**Comment by dscho** [comment]
FWIW if I use a valid host name to which I have credentials, I can reproduce the issue. But this is so far outside of Git for Windows' mission that I will not chase this bug.

If you want to, @cw789, feel free to

- install [Git for Windows' SDK](https://gitforwindows.org/#download-sdk),
- `sdk build openssh`
- `sdk cd openssh && cd src/openssh-*`
- then play around with either editing and recompiling, or with `gdb`, to find out what is going wrong.

**Comment by cw789** [comment]
@dscho: Thank you Johannes.

I've tried now to patch the sftp.c for the msys2 build.
But as I'm not skilled with C and building C - I was not successful and giving up for now.

By the way I don't see the great necessity as `sftp` within Win32-OpenSSH doesn't have this bug.

**Comment by dscho** [comment]
So let's close this here. If you are interested in seeing Git for Windows' `sftp` to be fixed, maybe you want to open an issue at https://github.com/msys2/MSYS2-packages.

**Comment by JulienCochuyt** [comment]
For the record, same issue/behavior here with MinTTY

git version 2.26.1.windows.1
cpu: x86_64
built from commit: 14af9f9d205f4a6a033c7b6f48980d1eeee2c610
sizeof-long: 4
sizeof-size_t: 8

Windows 10 Version 1909 (18363.778)

**Comment by dscho** [comment]
@JulienCochuyt [this comment](https://github.com/git-for-windows/git/issues/2257#issuecomment-510458311) applies to you, too.

**Comment by JulienCochuyt** [comment]
Fixed as of v2.29.2(3)
